### PR TITLE
Enable loop simplify pass for all llvm.loop attributes

### PR DESCRIPTION
**Pull request by vmaksimo**
It is done to avoid duplicate OpLoopMerge instruction in case of continue
operand in the loop.

### CODE REVIEW

File: lib/SPIRV/SPIRVInternal.h [hunk 1]
```diff
@@ -930,7 +930,7 @@ template <> inline void SPIRVMap<std::string, Op, SPIRVOpaqueType>::init() {
 }
 
 // Check if the module contains llvm.loop.unroll.* metadata
```
Comment (svenvh) [inline lib/SPIRV/SPIRVInternal.h]:
```suggestion
// Check if the module contains llvm.loop.* metadata
```

File: lib/SPIRV/SPIRVUtil.cpp [hunk 2]
```diff
@@ -1501,16 +1501,13 @@ llvm::Constant *getOCLNullClkEventPtr(Module *M) {
...
   for (const Function &F : *M)
     for (const BasicBlock &BB : F) {
       const Instruction *Term = BB.getTerminator();
       if (!Term)
         continue;
       if (const MDNode *MD = Term->getMetadata("llvm.loop"))
-        for (const MDOperand &MDOp : MD->operands())
-          if (getMDOperandAsString(dyn_cast<MDNode>(MDOp), 0)
-                  .find("llvm.loop.unroll.") == 0)
             return true;
```
Comment (mlychkov) [inline lib/SPIRV/SPIRVUtil.cpp]: Please align indent.

File: test/transcoding/FPGAInfiniteLoop.ll [hunk 3]
```diff
@@ -0,0 +1,178 @@
+; RUN: llvm-as %s -o %t.bc
```
Comment (AGindinson) [inline test/transcoding/FPGAInfiniteLoop.ll]: I'd suggest commenting out the source code and providing the link to the intel/llvm project, which apparently was used to produce the LLVM IR (the same way it was done for test/IntelFPGAReg.ll)

**Comment by vmaksimo** [comment]
> `while (true)`

Where is the difference? `while (i < m)` will always be true

**Comment by mlychkov** [comment]
> > `while (true)`
> 
> Where is the difference? `while (i < m)` will always be true

Looks like compiler doesn't detect that this is infinite loop.

File: lib/SPIRV/SPIRVWriter.cpp [hunk 4]
```diff
@@ -2154,7 +2154,7 @@ bool llvm::writeSpirv(Module *M, const SPIRV::TranslatorOpts &Opts,
 
   legacy::PassManager PassMgr;
   addPassesForSPIRV(PassMgr);
-  if (hasLoopUnrollMetadata(M))
+  if (hasLoopMetadata(M))
```
Comment (mlychkov) [inline lib/SPIRV/SPIRVWriter.cpp]: [Optional] It might be good to have a comment why we need loopsimplify pass here.

Comment (AlexeySotkin) [reply lib/SPIRV/SPIRVWriter.cpp]: Indeed. @vmaksimo please add a comment explaining why we need to simplify loops.

**Pull request merged**